src/applications/vaos/appointment-list/components/ConfirmedAppointmentDetailsPage/ConfirmedAppointmentDetailsPage.jsx

```diff
@@ -59,34 +69,45 @@ function ConfirmedAppointmentDetailsPage({
   const { id } = useParams();
   const history = useHistory();
 
-  const appointment = appointmentDetails?.[id];
-
   useEffect(() => {
     const status = confirmedStatus === FETCH_STATUS.succeeded;
 
     if (!status) {
       history.push('/');
     }
 
-    if (!appointment) {
+    if (!appointment || appointment.id !== id) {
       fetchConfirmedAppointmentDetails(id);
     }
 
     scrollAndFocus();
   }, []);
 
+  useEffect(
+    () => {
+      if (
+        !cancelInfo.showCancelModal &&
+        cancelInfo.cancelAppointmentStatus === FETCH_STATUS.succeeded
+      ) {
+        scrollAndFocus();
+      }
+    },
+    [cancelInfo.showCancelModal, cancelInfo.cancelAppointmentStatus],
+  );
+
   if (appointmentDetailsStatus === FETCH_STATUS.loading) {
     return (
       <div className="vads-u-margin-y--8">
-        <LoadingIndicator message="Loading your appointment request...!!!!" />
+        <LoadingIndicator message="Loading your appointment..." />
       </div>
     );
   }
 
-  if (!appointment) {
+  if (!appointment || appointment.id !== id) {
```

Comment:
Prevents rendering if `currentAppointment` in redux state hasn't been updated to the current appointment yet.